### ISSUE TITLE
MultilineBlockLayout with LineLength and without assignment

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop-hq/rubocop#7788]
We have #7284 with some fix.

In the tests there are `some_result = lambda do |`, and for such code there are no offenses.

But in the documentation from this PR there is `blah { |` (as "good"), without assignment, and such code still brings offense from `MultilineBlockLayout`.

--------

## Expected behavior

No offenses.

## Actual behavior

There is offense from `MultilineBlockLayout`.

## Steps to reproduce the problem

Just a code with multiline parameters, exceeding `LineLength`, and without assignment, line in the documentation with `blah { |`.

## RuboCop version

```
0.80.0 (using Parser 2.7.0.2, running on ruby 2.6.5 x86_64-linux)
```

**Comment by AlexWayfer** [comment]
Sorry, forgot to mention @jonas054 in case if they're interested.

**Comment by mvz** [comment]
Can you add a concrete example of the whole block where you see this problem?

**Comment by AlexWayfer** [comment]
> Can you add a concrete example of the whole block where you see this problem?

As I wrote: the documentation (`blah { |`) from this PR is not "good", as wrote in this PR, and make offenses.

Example of the same code with resolved other default cops:

```ruby
# test.rb
# frozen_string_literal: true

blah do |
  _long_list,
  _of_parameters,
  _that_would_not,
  _fit_on_one_line
|
  foo(i)
  bar(i)
end
```

```
> rubocop test.rb
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
 - Style/HashEachMethods (0.80)
 - Style/HashTransformKeys (0.80)
 - Style/HashTransformValues (0.80)
For more information: https://docs.rubocop.org/en/latest/versioning/
Inspecting 1 file
C

Offenses:

test.rb:3:9: C: Layout/MultilineBlockLayout: Block argument expression is not on the same line as the block start.
blah do | ...
        ^

1 file inspected, 1 offense detected
```

```
> rubocop -V
0.80.1 (using Parser 2.7.0.4, running on ruby 2.7.0 x86_64-linux)
```

The difference, as I see, in an assignment.

**Comment by mvz** [comment]
Thanks, @AlexWayfer, having the whole example here helps clear up what the issue is.

**Comment by jonas054** [comment]
I can't reproduce the problem. With the example you gave, @AlexWayfer, I get the same offense report, but only because the argument list _does_ fit on a single line. If I make it longer the problem goes away.

**Comment by AlexWayfer** [comment]
> I can't reproduce the problem. With the example you gave, @AlexWayfer, I get the same offense report, but only because the argument list _does_ fit on a single line. If I make it longer the problem goes away.

That's true. I can't reproduce it locally too. I don't remember a project with this issue. So, I'm going to close this and I'll reopen if I'll have more details.

Sorry to bothering you.
